src/services/completions.ts

```diff
@@ -1501,13 +1501,19 @@ namespace ts.Completions {
                     : KeywordCompletionFilters.TypeKeywords;
             }
 
+            const undeclaredVariable = findUndeclaredVariableIfExist(location);
+
             filterMutate(symbols, symbol => {
                 if (!isSourceFile(location)) {
                     // export = /**/ here we want to get all meanings, so any symbol is ok
                     if (isExportAssignment(location.parent)) {
                         return true;
                     }
 
+                    if (isUndeclaredSymbol(undeclaredVariable, symbol)) {

```

Comment:
```suggestion
                    // Filter out variables from their own initializers, e.g. `const o = { prop: /* no 'o' here */ }`
                    if (objectLiteralVariableDeclaration && contains(symbol.declarations, objectLiteralVariableDeclaration)) {
```

Reply:
Actually, in these cases the declaration should always be the valueDeclaration, right? This is a bit more efficient and I think should work:

```suggestion
                    if (objectLiteralVariableDeclaration && symbol.valueDeclaration === objectLiteralVariableDeclaration) {
```